src/utils/deepClone.ts

```diff
@@ -0,0 +1,16 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+export const deepClone = (_o: any): any =>
+        _o && typeof _o === "object"
+            ? Array.isArray(_o)
+                ? _o.map((_value) => deepClone(_value))
+                : Object.fromEntries(
+                    Object.entries(_o).map(
+                        ([_key, _value]) =>
+                            [_key, deepClone(_value)]))
+            : _o;
```

Comment:
Just a note: this last segment of the top-level ternary operator captures all "falsy" `_o` values (which is fine, but worth noting, and perhaps worth commenting in the code, for the avoidance of doubt for future readers / code reviewers): `undefined`, `null`, `""`, `[]`, `0` (as well as any "truthy" value that is not an `object` (including `Array`)).

Reply:
I added a function helper.
feel free to correct it